rust/lancedb/src/remote/table.rs

```diff
@@ -266,11 +402,79 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
         self.check_table_response(response).await?;
         Ok(())
     }
-    async fn create_index(&self, _index: IndexBuilder) -> Result<()> {
-        Err(Error::NotSupported {
-            message: "create_index is not yet supported on LanceDB cloud.".into(),
-        })
+
+    async fn create_index(&self, mut index: IndexBuilder) -> Result<()> {
+        let request = self
+            .client
+            .post(&format!("/table/{}/create_index/", self.name));
+
+        let column = match index.columns.len() {
+            0 => {
+                return Err(Error::InvalidInput {
+                    message: "No columns specified".into(),
+                })
+            }
+            1 => index.columns.pop().unwrap(),
+            _ => {
+                return Err(Error::NotSupported {
+                    message: "Indices over multiple columns not yet supported".into(),
+                })
+            }
+        };
+        let mut body = serde_json::json!({
+            "column": column
+        });
+
+        let (index_type, distance_type) = match index.index {
+            // TODO: Should we pass the actual index parameters? SaaS does not
+            // yet support them.
+            Index::IvfPq(index) => ("IVF_PQ", Some(index.distance_type)),
+            Index::IvfHnswSq(index) => ("IVF_HNSW_SQ", Some(index.distance_type)),
+            Index::BTree(_) => ("BTREE", None),
+            Index::Bitmap(_) => ("BITMAP", None),
+            Index::LabelList(_) => ("LABEL_LIST", None),
+            Index::FTS(_) => ("FTS", None),
+            Index::Auto => {
+                let schema = self.schema().await?;
+                let field = schema
+                    .field_with_name(&column)
+                    .map_err(|_| Error::InvalidInput {
+                        message: format!("Column {} not found in schema", column),
+                    })?;
+                if supported_vector_data_type(field.data_type()) {
+                    ("IVF_PQ", None)
+                } else if supported_btree_data_type(field.data_type()) {
+                    ("BTREE", None)
+                } else {
+                    return Err(Error::NotSupported {
+                        message: format!(
+                            "there are no indices supported for the field `{}` with the data type {}",
+                            field.name(),
+                            field.data_type()
+                        ),
+                    });
+                }
+            }
+            _ => {
+                return Err(Error::NotSupported {
+                    message: "Index type not supported".into(),
+                })
+            }
+        };
+        body["index_type"] = serde_json::Value::String(index_type.into());
+        if let Some(distance_type) = distance_type {
+            body["distance_type"] = serde_json::Value::String(distance_type.to_string());
+        }
```

Comment:
Didn't you calculate `distance_type` above in the `match index.index` block?

Reply:
I did, but here I am adding it to the body.